In [1]:
from sklearn.ensemble import AdaBoostClassifier, GradientBoostingClassifier, RandomForestClassifier
from sklearn.tree import DecisionTreeClassifier
from sklearn.neighbors import KNeighborsClassifier
from sklearn.linear_model import LogisticRegression
from sklearn.naive_bayes import GaussianNB
from sklearn.svm import SVC
from sklearn.neural_network import MLPClassifier
from xgboost import XGBClassifier
from eipy.ei import EnsembleIntegration
import eipy.utils as ut
from eipy.additional_ensembles import MeanAggregation, CES
from sklearn.model_selection import train_test_split
import numpy as np
import pandas as pd
from sklearn import datasets
import pickle as pkl
pd.set_option('display.max_columns', None)

In [2]:
data = open("/home/opc/block_vol/tadpole/tadpole_data_imptn_norm.pickle", "rb")
data = pkl.load(data)
data["cn_bl"]

{'train': {'main_cognitive_tests': array([[0.        , 0.21052632, 0.29166667, ..., 0.4       , 0.6363635 ,
          0.        ],
         [0.        , 0.47368421, 0.5       , ..., 0.5       , 0.7272725 ,
          0.        ],
         [0.        , 0.35105263, 0.40291667, ..., 0.2       , 0.46153845,
          0.        ],
         ...,
         [0.        , 0.26315789, 0.375     , ..., 0.45      , 0.75      ,
          0.        ],
         [0.        , 0.15789474, 0.20833333, ..., 0.6       , 0.769231  ,
          0.        ],
         [0.        , 0.45631579, 0.61125   , ..., 0.45      , 0.75      ,
          0.        ]]),
  'mri_vols': array([[0.45837351, 0.40070269, 0.3679434 , ..., 0.59772024, 0.48045342,
          0.47554905],
         [0.47568581, 0.37443534, 0.50583728, ..., 0.73337359, 0.51780264,
          0.608286  ],
         [0.2447696 , 0.11360214, 0.        , ..., 0.31637352, 0.20597781,
          0.15754504],
         ...,
         [0.45602038, 0.72678601, 0.6148599

In [3]:
X_train=data["cn_bl"]["train"]

In [4]:
X_test=data["cn_bl"]["test"]

In [5]:
y_train = pd.read_csv("/home/opc/block_vol/tadpole/tadpole_training_labels.csv")
y_train.drop(columns=["Unnamed: 0"], axis=1, inplace=True)
y_train = y_train["last_dx"]
y_train.value_counts()/y_train.shape[0]

last_dx
CN     0.827586
MCI    0.123153
DEM    0.049261
Name: count, dtype: float64

In [6]:
y_test = pd.read_csv("/home/opc/block_vol/tadpole/tadpole_testing_labels.csv")
y_test.drop(columns=["Unnamed: 0"], axis=1, inplace=True)
y_test = y_test["last_dx"]


In [7]:
encoding_dict = {'CN': 0, 'MCI': 1, 'DEM': 2}
y_train = y_train.map(encoding_dict)
y_test = y_test.map(encoding_dict)

In [8]:
base_predictors = {
                    'ADAB': AdaBoostClassifier(),
                    'XGB': XGBClassifier(),
                    'DT': DecisionTreeClassifier(),
                    'RF': RandomForestClassifier(),
                    'GB': GradientBoostingClassifier(),
                    'KNN': KNeighborsClassifier(),
                    'LR': LogisticRegression(multi_class="auto", solver="lbfgs"),
                    'NB': GaussianNB(),
                    'MLP': MLPClassifier(),
                    'SVM': SVC(probability=True)
}

In [9]:
EI = EnsembleIntegration(
                        base_predictors=base_predictors,
                        k_outer=5,
                        k_inner=5,
                        n_samples=1,
                        sampling_strategy="undersampling",
                        sampling_aggregation=None,
                        n_jobs=-1,
                        random_state=42,
                        project_name="tadpole",
                        model_building=True,
                        )

In [10]:
EI.fit_base(X_train, y_train)

Training base predictors on main_cognitive_tests...
        
... for ensemble performance analysis...


Generating meta training data: |██████████|100%
Generating meta test data: |██████████|100%



... for final ensemble...


Generating meta training data: |██████████|100%
Training final base predictors: |██████████|100%




Training base predictors on mri_vols...
        
... for ensemble performance analysis...


Generating meta training data: |██████████|100%
Generating meta test data: |██████████|100%



... for final ensemble...


Generating meta training data: |██████████|100%
Training final base predictors: |██████████|100%




Training base predictors on csf_bio...
        
... for ensemble performance analysis...


Generating meta training data: |██████████|100%
Generating meta test data: |██████████|100%



... for final ensemble...


Generating meta training data: |██████████|100%
Training final base predictors: |██████████|100%




Training base predictors on other...
        
... for ensemble performance analysis...


Generating meta training data: |██████████|100%
Generating meta test data: |██████████|100%



... for final ensemble...


Generating meta training data: |██████████|100%
Training final base predictors: |██████████|100%




Training base predictors on MRI ROI: Volume (WM Parcellation)...
        
... for ensemble performance analysis...


Generating meta training data: |██████████|100%
Generating meta test data: |██████████|100%



... for final ensemble...


Generating meta training data: |██████████|100%
Training final base predictors: |██████████|100%




Training base predictors on MRI ROI: Volume (Cortical Parcellation)...
        
... for ensemble performance analysis...


Generating meta training data: |██████████|100%
Generating meta test data: |██████████|100%



... for final ensemble...


Generating meta training data: |██████████|100%
Training final base predictors: |██████████|100%




Training base predictors on MRI ROI: Surface Area...
        
... for ensemble performance analysis...


Generating meta training data: |██████████|100%
Generating meta test data: |██████████|100%



... for final ensemble...


Generating meta training data: |██████████|100%
Training final base predictors: |██████████|100%




Training base predictors on MRI ROI: Cortical Thickness Average...
        
... for ensemble performance analysis...


Generating meta training data: |██████████|100%
Generating meta test data: |██████████|100%



... for final ensemble...


Generating meta training data: |██████████|100%
Training final base predictors: |██████████|100%




Training base predictors on MRI ROI: Cortical Thickness Standard Deviation...
        
... for ensemble performance analysis...


Generating meta training data: |██████████|100%
Generating meta test data: |██████████|100%



... for final ensemble...


Generating meta training data: |██████████|100%
Training final base predictors: |██████████|100%


In [11]:
ut.predictive_multiclass_data(EI.meta_training_data)
EI.fit_meta(meta_predictors=base_predictors)

Analyzing ensembles: |██████████|100%
Training final meta models: |██████████|100%


In [12]:
argmax = lambda x: np.argmax(x)
cols_to_transform = [col for col in EI.meta_predictions.columns if col != 'labels']
for column in cols_to_transform:
    EI.meta_predictions[column] = EI.meta_predictions[column].apply(argmax)
EI.meta_predictions

ADAB  XGB  DT  RF  GB  KNN  LR  NB  MLP  SVM  labels
0       0    0   1   0   0    1   0   2    0    0       0
1       0    0   0   0   0    0   0   0    0    0       0
2       0    0   1   0   0    0   0   0    0    0       2
3       0    0   2   0   0    0   0   2    0    0       0
4       0    0   0   0   0    0   0   0    0    0       0
..    ...  ...  ..  ..  ..  ...  ..  ..  ...  ...     ...
401     0    0   0   0   0    0   0   0    0    0       0
402     0    0   0   0   0    0   0   0    0    0       0
403     0    0   0   0   0    0   0   0    0    0       0
404     0    0   0   0   0    0   0   0    0    0       0
405     0    0   1   0   0    0   0   1    1    0       2

[406 rows x 11 columns]

In [13]:
df = EI.meta_predictions
model_columns = df.columns[:-1]

accuracy_dict = {}

for model in model_columns:
    correct_predictions = (df[model] == df["labels"]).sum()
    total_predictions = len(df)
    accuracy = correct_predictions / total_predictions
    accuracy_dict[model] = accuracy

accuracy_dict
accuracies = pd.Series(accuracy_dict)
accuracies

ADAB    0.716749
XGB     0.800493
DT      0.633005
RF      0.825123
GB      0.790640
KNN     0.815271
LR      0.758621
NB      0.628079
MLP     0.805419
SVM     0.825123
dtype: float64

In [14]:
preferred_meta_model = max(accuracy_dict, key=lambda key: accuracy_dict[key])
y_pred = EI.predict(X_dict=X_test, meta_model_key=preferred_meta_model)
y_pred = [np.argmax(np.array(y)) for y in y_pred]

In [15]:
accuracy = sum([1*(y==y_hat)+0*(y!=y_hat) for y,y_hat in list(zip(y_test, y_pred))])/len(y_test)
accuracy*100

82.35294117647058

In [16]:
accuracies = pd.DataFrame(accuracies)
accuracies.rename(columns={0:"accuracy"}, inplace=True)
accuracies = accuracies.T
metrics = pd.concat([EI.meta_summary["metrics"], accuracies])
metrics


ADAB       XGB        DT        RF        GB       KNN  \
precision  0.322663  0.342029  0.338683  0.275720  0.322382  0.342356   
recall     0.328413  0.333770  0.338730  0.332341  0.329802  0.334048   
f1         0.325508  0.318417  0.332851  0.301395  0.314978  0.311441   
accuracy   0.716749  0.800493  0.633005  0.825123  0.790640  0.815271   

                 LR        NB       MLP       SVM  
precision  0.382040  0.380528  0.491791  0.275720  
recall     0.366627  0.393770  0.385476  0.332341  
f1         0.369651  0.366258  0.396002  0.301395  
accuracy   0.758621  0.628079  0.805419  0.825123

Without a sampling strategy the accuracy is 82.35%.
When using undersampling, the accuracy is 82.35%.
When using oversampling, the accuracy is 81.37%.

In [17]:
sampling_strategies = ["undersampling", "oversampling"] #hybrid as well
metrics = ["accuracy", "precision", "recall", "f1"]

performance_frame = pd.DataFrame()

for strat in sampling_strategies:
    strat_scores = []
    for metric in metrics:
        EI = EnsembleIntegration(
                        base_predictors=base_predictors,
                        k_outer=5,
                        k_inner=5,
                        n_samples=1,
                        sampling_strategy=strat,
                        sampling_aggregation=None,
                        n_jobs=-1,
                        random_state=42,
                        project_name="tadpole",
                        model_building=True,
                        )
        print(f"EI for {strat} and scoring with {metric}")
        EI.fit_base(X_train, y_train)

        ut.predictive_multiclass_data(EI.meta_training_data)
        EI.fit_meta(meta_predictors=base_predictors)

        argmax = lambda x: np.argmax(x)
sampling_strategies = ["undersampling", "oversampling"] #hybrid as well
metrics = ["accuracy", "precision", "recall", "f1"]

performance_frame = pd.DataFrame()

for strat in sampling_strategies:
    strat_scores = []
    for metric in metrics:
        EI = EnsembleIntegration(
                        base_predictors=base_predictors,
                        k_outer=5,
                        k_inner=5,
                        n_samples=1,
                        sampling_strategy=strat,
                        sampling_aggregation=None,
                        n_jobs=-1,
                        random_state=42,
                        project_name="tadpole",
                        model_building=True,
                        )
        print(f"EI for {strat} and scoring with {metric}")
        EI.fit_base(X_train, y_train)

        ut.predictive_multiclass_data(EI.meta_training_data)
        EI.fit_meta(meta_predictors=base_predictors)

        argmax = lambda x: np.argmax(x)
        cols_to_transform = [col for col in EI.meta_predictions.columns if col != 'labels']
        for column in cols_to_transform:
            EI.meta_predictions[column] = EI.meta_predictions[column].apply(argmax)
        
        if metric == "accuracy":
            df = EI.meta_predictions
            model_columns = df.columns[:-1]
            accuracies = {}
            for model in model_columns:
                correct_predictions = (df[model] == df["labels"]).sum()
                total_predictions = len(df)
                accuracy = correct_predictions / total_predictions
                accuracies[model] = accuracy
            accuracies = pd.Series(accuracies)
            accuracies = pd.DataFrame(accuracies)
            accuracies.rename(columns={0:"accuracy"}, inplace=True)
            accuracies = accuracies.T
            EI.meta_summary["metrics"] = pd.concat([EI.meta_summary["metrics"], accuracies])
        

        preferred_meta_model = EI.meta_summary["metrics"].loc[metric].idxmax()
        y_pred = EI.predict(X_dict=X_test, meta_model_key=preferred_meta_model)
        y_pred = [np.argmax(np.array(y)) for y in y_pred]

        if metric == "accuracy":
            strat_scores.append(sum([1*(y==y_hat)+0*(y!=y_hat) for y,y_hat in list(zip(y_test, y_pred))])/len(y_test))
        elif metric == "precision":
            strat_scores.append(ut.precision_score(y_test, y_pred, average='macro'))
        elif metric == "recall":
            strat_scores.append(ut.recall_score(y_test, y_pred, average='macro'))
        elif metric == "f1":
            strat_scores.append(ut.f1_score(y_test, y_pred, average='macro'))
        
    performance_frame[strat] = strat_scores    

EI for undersampling and scoring with accuracy
Training base predictors on main_cognitive_tests...
        
... for ensemble performance analysis...


Generating meta training data: |██████████|100%
Generating meta test data: |██████████|100%



... for final ensemble...


Generating meta training data: |██████████|100%
Training final base predictors: |██████████|100%




Training base predictors on mri_vols...
        
... for ensemble performance analysis...


Generating meta training data: |██████████|100%
Generating meta test data: |██████████|100%



... for final ensemble...


Generating meta training data: |██████████|100%
Training final base predictors: |██████████|100%




Training base predictors on csf_bio...
        
... for ensemble performance analysis...


Generating meta training data: |██████████|100%
Generating meta test data: |██████████|100%



... for final ensemble...


Generating meta training data: |██████████|100%
Training final base predictors: |██████████|100%




Training base predictors on other...
        
... for ensemble performance analysis...


Generating meta training data: |██████████|100%
Generating meta test data: |██████████|100%



... for final ensemble...


Generating meta training data: |██████████|100%
Training final base predictors: |██████████|100%




Training base predictors on MRI ROI: Volume (WM Parcellation)...
        
... for ensemble performance analysis...


Generating meta training data: |██████████|100%
Generating meta test data: |██████████|100%



... for final ensemble...


Generating meta training data: |██████████|100%
Training final base predictors: |██████████|100%




Training base predictors on MRI ROI: Volume (Cortical Parcellation)...
        
... for ensemble performance analysis...


Generating meta training data: |██████████|100%
Generating meta test data: |██████████|100%



... for final ensemble...


Generating meta training data: |██████████|100%
Training final base predictors: |██████████|100%




Training base predictors on MRI ROI: Surface Area...
        
... for ensemble performance analysis...


Generating meta training data: |██████████|100%
Generating meta test data: |██████████|100%



... for final ensemble...


Generating meta training data: |██████████|100%
Training final base predictors: |██████████|100%




Training base predictors on MRI ROI: Cortical Thickness Average...
        
... for ensemble performance analysis...


Generating meta training data: |██████████|100%
Generating meta test data: |██████████|100%



... for final ensemble...


Generating meta training data: |██████████|100%
Training final base predictors: |██████████|100%




Training base predictors on MRI ROI: Cortical Thickness Standard Deviation...
        
... for ensemble performance analysis...


Generating meta training data: |██████████|100%
Generating meta test data: |██████████|100%



... for final ensemble...


Generating meta training data: |██████████|100%
Training final base predictors: |██████████|100%


Analyzing ensembles: |██████████|100%
Training final meta models: |██████████|100%


EI for undersampling and scoring with precision
Training base predictors on main_cognitive_tests...
        
... for ensemble performance analysis...


Generating meta training data: |██████████|100%
Generating meta test data: |██████████|100%



... for final ensemble...


Generating meta training data: |██████████|100%
Training final base predictors: |██████████|100%




Training base predictors on mri_vols...
        
... for ensemble performance analysis...


Generating meta training data: |██████████|100%
Generating meta test data: |██████████|100%



... for final ensemble...


Generating meta training data: |██████████|100%
Training final base predictors: |██████████|100%




Training base predictors on csf_bio...
        
... for ensemble performance analysis...


Generating meta training data: |██████████|100%
Generating meta test data: |██████████|100%



... for final ensemble...


Generating meta training data: |██████████|100%
Training final base predictors: |██████████|100%




Training base predictors on other...
        
... for ensemble performance analysis...


Generating meta training data: |██████████|100%
Generating meta test data: |██████████|100%



... for final ensemble...


Generating meta training data: |██████████|100%
Training final base predictors: |██████████|100%




Training base predictors on MRI ROI: Volume (WM Parcellation)...
        
... for ensemble performance analysis...


Generating meta training data: |██████████|100%
Generating meta test data: |██████████|100%



... for final ensemble...


Generating meta training data: |██████████|100%
Training final base predictors: |██████████|100%




Training base predictors on MRI ROI: Volume (Cortical Parcellation)...
        
... for ensemble performance analysis...


Generating meta training data: |██████████|100%
Generating meta test data: |██████████|100%



... for final ensemble...


Generating meta training data: |██████████|100%
Training final base predictors: |██████████|100%




Training base predictors on MRI ROI: Surface Area...
        
... for ensemble performance analysis...


Generating meta training data: |██████████|100%
Generating meta test data: |██████████|100%



... for final ensemble...


Generating meta training data: |██████████|100%
Training final base predictors: |██████████|100%




Training base predictors on MRI ROI: Cortical Thickness Average...
        
... for ensemble performance analysis...


Generating meta training data: |██████████|100%
Generating meta test data: |██████████|100%



... for final ensemble...


Generating meta training data: |██████████|100%
Training final base predictors: |██████████|100%




Training base predictors on MRI ROI: Cortical Thickness Standard Deviation...
        
... for ensemble performance analysis...


Generating meta training data: |██████████|100%
Generating meta test data: |██████████|100%



... for final ensemble...


Generating meta training data: |██████████|100%
Training final base predictors: |██████████|100%


Analyzing ensembles: |██████████|100%
Training final meta models: |██████████|100%


EI for undersampling and scoring with recall
Training base predictors on main_cognitive_tests...
        
... for ensemble performance analysis...


Generating meta training data: |██████████|100%
Generating meta test data: |██████████|100%



... for final ensemble...


Generating meta training data: |██████████|100%
Training final base predictors: |██████████|100%




Training base predictors on mri_vols...
        
... for ensemble performance analysis...


Generating meta training data: |██████████|100%
Generating meta test data: |██████████|100%



... for final ensemble...


Generating meta training data: |██████████|100%
Training final base predictors: |██████████|100%




Training base predictors on csf_bio...
        
... for ensemble performance analysis...


Generating meta training data: |██████████|100%
Generating meta test data: |██████████|100%



... for final ensemble...


Generating meta training data: |██████████|100%
Training final base predictors: |██████████|100%




Training base predictors on other...
        
... for ensemble performance analysis...


Generating meta training data: |██████████|100%
Generating meta test data: |██████████|100%



... for final ensemble...


Generating meta training data: |██████████|100%
Training final base predictors: |██████████|100%




Training base predictors on MRI ROI: Volume (WM Parcellation)...
        
... for ensemble performance analysis...


Generating meta training data: |██████████|100%
Generating meta test data: |██████████|100%



... for final ensemble...


Generating meta training data: |██████████|100%
Training final base predictors: |██████████|100%




Training base predictors on MRI ROI: Volume (Cortical Parcellation)...
        
... for ensemble performance analysis...


Generating meta training data: |██████████|100%
Generating meta test data: |██████████|100%



... for final ensemble...


Generating meta training data: |██████████|100%
Training final base predictors: |██████████|100%




Training base predictors on MRI ROI: Surface Area...
        
... for ensemble performance analysis...


Generating meta training data: |██████████|100%
Generating meta test data: |██████████|100%



... for final ensemble...


Generating meta training data: |██████████|100%
Training final base predictors: |██████████|100%




Training base predictors on MRI ROI: Cortical Thickness Average...
        
... for ensemble performance analysis...


Generating meta training data: |██████████|100%
Generating meta test data: |██████████|100%



... for final ensemble...


Generating meta training data: |██████████|100%
Training final base predictors: |██████████|100%




Training base predictors on MRI ROI: Cortical Thickness Standard Deviation...
        
... for ensemble performance analysis...


Generating meta training data: |██████████|100%
Generating meta test data: |██████████|100%



... for final ensemble...


Generating meta training data: |██████████|100%
Training final base predictors: |██████████|100%


Analyzing ensembles: |██████████|100%
Training final meta models: |██████████|100%


EI for undersampling and scoring with f1
Training base predictors on main_cognitive_tests...
        
... for ensemble performance analysis...


Generating meta training data: |██████████|100%
Generating meta test data: |██████████|100%



... for final ensemble...


Generating meta training data: |██████████|100%
Training final base predictors: |██████████|100%




Training base predictors on mri_vols...
        
... for ensemble performance analysis...


Generating meta training data: |██████████|100%
Generating meta test data: |██████████|100%



... for final ensemble...


Generating meta training data: |██████████|100%
Training final base predictors: |██████████|100%




Training base predictors on csf_bio...
        
... for ensemble performance analysis...


Generating meta training data: |██████████|100%
Generating meta test data: |██████████|100%



... for final ensemble...


Generating meta training data: |██████████|100%
Training final base predictors: |██████████|100%




Training base predictors on other...
        
... for ensemble performance analysis...


Generating meta training data: |██████████|100%
Generating meta test data: |██████████|100%



... for final ensemble...


Generating meta training data: |██████████|100%
Training final base predictors: |██████████|100%




Training base predictors on MRI ROI: Volume (WM Parcellation)...
        
... for ensemble performance analysis...


Generating meta training data: |██████████|100%
Generating meta test data: |██████████|100%



... for final ensemble...


Generating meta training data: |██████████|100%
Training final base predictors: |██████████|100%




Training base predictors on MRI ROI: Volume (Cortical Parcellation)...
        
... for ensemble performance analysis...


Generating meta training data: |██████████|100%
Generating meta test data: |██████████|100%



... for final ensemble...


Generating meta training data: |██████████|100%
Training final base predictors: |██████████|100%




Training base predictors on MRI ROI: Surface Area...
        
... for ensemble performance analysis...


Generating meta training data: |██████████|100%
Generating meta test data: |██████████|100%



... for final ensemble...


Generating meta training data: |██████████|100%
Training final base predictors: |██████████|100%




Training base predictors on MRI ROI: Cortical Thickness Average...
        
... for ensemble performance analysis...


Generating meta training data: |██████████|100%
Generating meta test data: |██████████|100%



... for final ensemble...


Generating meta training data: |██████████|100%
Training final base predictors: |██████████|100%




Training base predictors on MRI ROI: Cortical Thickness Standard Deviation...
        
... for ensemble performance analysis...


Generating meta training data: |██████████|100%
Generating meta test data: |██████████|100%



... for final ensemble...


Generating meta training data: |██████████|100%
Training final base predictors: |██████████|100%


Analyzing ensembles: |██████████|100%
Training final meta models: |██████████|100%


EI for oversampling and scoring with accuracy
Training base predictors on main_cognitive_tests...
        
... for ensemble performance analysis...


Generating meta training data: |██████████|100%
Generating meta test data: |██████████|100%



... for final ensemble...


Generating meta training data: |██████████|100%
Training final base predictors: |██████████|100%




Training base predictors on mri_vols...
        
... for ensemble performance analysis...


Generating meta training data: |██████████|100%
Generating meta test data: |██████████|100%



... for final ensemble...


Generating meta training data: |██████████|100%
Training final base predictors: |██████████|100%




Training base predictors on csf_bio...
        
... for ensemble performance analysis...


Generating meta training data: |██████████|100%
Generating meta test data: |██████████|100%



... for final ensemble...


Generating meta training data: |██████████|100%
Training final base predictors: |██████████|100%




Training base predictors on other...
        
... for ensemble performance analysis...


Generating meta training data: |██████████|100%
Generating meta test data: |██████████|100%



... for final ensemble...


Generating meta training data: |██████████|100%
Training final base predictors: |██████████|100%




Training base predictors on MRI ROI: Volume (WM Parcellation)...
        
... for ensemble performance analysis...


Generating meta training data: |██████████|100%
Generating meta test data: |██████████|100%



... for final ensemble...


Generating meta training data: |██████████|100%
Training final base predictors: |██████████|100%




Training base predictors on MRI ROI: Volume (Cortical Parcellation)...
        
... for ensemble performance analysis...


Generating meta training data: |██████████|100%
Generating meta test data: |██████████|100%



... for final ensemble...


Generating meta training data: |██████████|100%
Training final base predictors: |██████████|100%




Training base predictors on MRI ROI: Surface Area...
        
... for ensemble performance analysis...


Generating meta training data: |██████████|100%
Generating meta test data: |██████████|100%



... for final ensemble...


Generating meta training data: |██████████|100%
Training final base predictors: |██████████|100%




Training base predictors on MRI ROI: Cortical Thickness Average...
        
... for ensemble performance analysis...


Generating meta training data: |██████████|100%
Generating meta test data: |██████████|100%



... for final ensemble...


Generating meta training data: |██████████|100%
Training final base predictors: |██████████|100%




Training base predictors on MRI ROI: Cortical Thickness Standard Deviation...
        
... for ensemble performance analysis...


Generating meta training data: |██████████|100%
Generating meta test data: |██████████|100%



... for final ensemble...


Generating meta training data: |██████████|100%
Training final base predictors: |██████████|100%


Analyzing ensembles: |██████████|100%
Training final meta models: |██████████|100%


EI for oversampling and scoring with precision
Training base predictors on main_cognitive_tests...
        
... for ensemble performance analysis...


Generating meta training data: |██████████|100%
Generating meta test data: |██████████|100%



... for final ensemble...


Generating meta training data: |██████████|100%
Training final base predictors: |██████████|100%




Training base predictors on mri_vols...
        
... for ensemble performance analysis...


Generating meta training data: |██████████|100%
Generating meta test data: |██████████|100%



... for final ensemble...


Generating meta training data: |██████████|100%
Training final base predictors: |██████████|100%




Training base predictors on csf_bio...
        
... for ensemble performance analysis...


Generating meta training data: |██████████|100%
Generating meta test data: |██████████|100%



... for final ensemble...


Generating meta training data: |██████████|100%
Training final base predictors: |██████████|100%




Training base predictors on other...
        
... for ensemble performance analysis...


Generating meta training data: |██████████|100%
Generating meta test data: |██████████|100%



... for final ensemble...


Generating meta training data: |██████████|100%
Training final base predictors: |██████████|100%




Training base predictors on MRI ROI: Volume (WM Parcellation)...
        
... for ensemble performance analysis...


Generating meta training data: |██████████|100%
Generating meta test data: |██████████|100%



... for final ensemble...


Generating meta training data: |██████████|100%
Training final base predictors: |██████████|100%




Training base predictors on MRI ROI: Volume (Cortical Parcellation)...
        
... for ensemble performance analysis...


Generating meta training data: |██████████|100%
Generating meta test data: |██████████|100%



... for final ensemble...


Generating meta training data: |██████████|100%
Training final base predictors: |██████████|100%




Training base predictors on MRI ROI: Surface Area...
        
... for ensemble performance analysis...


Generating meta training data: |██████████|100%
Generating meta test data: |██████████|100%



... for final ensemble...


Generating meta training data: |██████████|100%
Training final base predictors: |██████████|100%




Training base predictors on MRI ROI: Cortical Thickness Average...
        
... for ensemble performance analysis...


Generating meta training data: |██████████|100%
Generating meta test data: |██████████|100%



... for final ensemble...


Generating meta training data: |██████████|100%
Training final base predictors: |██████████|100%




Training base predictors on MRI ROI: Cortical Thickness Standard Deviation...
        
... for ensemble performance analysis...


Generating meta training data: |██████████|100%
Generating meta test data: |██████████|100%



... for final ensemble...


Generating meta training data: |██████████|100%
Training final base predictors: |██████████|100%


Analyzing ensembles: |██████████|100%
Training final meta models: |██████████|100%


EI for oversampling and scoring with recall
Training base predictors on main_cognitive_tests...
        
... for ensemble performance analysis...


Generating meta training data: |██████████|100%
Generating meta test data: |██████████|100%



... for final ensemble...


Generating meta training data: |██████████|100%
Training final base predictors: |██████████|100%




Training base predictors on mri_vols...
        
... for ensemble performance analysis...


Generating meta training data: |██████████|100%
Generating meta test data: |██████████|100%



... for final ensemble...


Generating meta training data: |██████████|100%
Training final base predictors: |██████████|100%




Training base predictors on csf_bio...
        
... for ensemble performance analysis...


Generating meta training data: |██████████|100%
Generating meta test data: |██████████|100%



... for final ensemble...


Generating meta training data: |██████████|100%
Training final base predictors: |██████████|100%




Training base predictors on other...
        
... for ensemble performance analysis...


Generating meta training data: |██████████|100%
Generating meta test data: |██████████|100%



... for final ensemble...


Generating meta training data: |██████████|100%
Training final base predictors: |██████████|100%




Training base predictors on MRI ROI: Volume (WM Parcellation)...
        
... for ensemble performance analysis...


Generating meta training data: |██████████|100%
Generating meta test data: |██████████|100%



... for final ensemble...


Generating meta training data: |██████████|100%
Training final base predictors: |██████████|100%




Training base predictors on MRI ROI: Volume (Cortical Parcellation)...
        
... for ensemble performance analysis...


Generating meta training data: |██████████|100%
Generating meta test data: |██████████|100%



... for final ensemble...


Generating meta training data: |██████████|100%
Training final base predictors: |██████████|100%




Training base predictors on MRI ROI: Surface Area...
        
... for ensemble performance analysis...


Generating meta training data: |██████████|100%
Generating meta test data: |██████████|100%



... for final ensemble...


Generating meta training data: |██████████|100%
Training final base predictors: |██████████|100%




Training base predictors on MRI ROI: Cortical Thickness Average...
        
... for ensemble performance analysis...


Generating meta training data: |██████████|100%
Generating meta test data: |██████████|100%



... for final ensemble...


Generating meta training data: |██████████|100%
Training final base predictors: |██████████|100%




Training base predictors on MRI ROI: Cortical Thickness Standard Deviation...
        
... for ensemble performance analysis...


Generating meta training data: |██████████|100%
Generating meta test data: |██████████|100%



... for final ensemble...


Generating meta training data: |██████████|100%
Training final base predictors: |██████████|100%


Analyzing ensembles: |██████████|100%
Training final meta models: |██████████|100%


EI for oversampling and scoring with f1
Training base predictors on main_cognitive_tests...
        
... for ensemble performance analysis...


Generating meta training data: |██████████|100%
Generating meta test data: |██████████|100%



... for final ensemble...


Generating meta training data: |██████████|100%
Training final base predictors: |██████████|100%




Training base predictors on mri_vols...
        
... for ensemble performance analysis...


Generating meta training data: |██████████|100%
Generating meta test data: |██████████|100%



... for final ensemble...


Generating meta training data: |██████████|100%
Training final base predictors: |██████████|100%




Training base predictors on csf_bio...
        
... for ensemble performance analysis...


Generating meta training data: |██████████|100%
Generating meta test data: |██████████|100%



... for final ensemble...


Generating meta training data: |██████████|100%
Training final base predictors: |██████████|100%




Training base predictors on other...
        
... for ensemble performance analysis...


Generating meta training data: |██████████|100%
Generating meta test data: |██████████|100%



... for final ensemble...


Generating meta training data: |██████████|100%
Training final base predictors: |██████████|100%




Training base predictors on MRI ROI: Volume (WM Parcellation)...
        
... for ensemble performance analysis...


Generating meta training data: |██████████|100%
Generating meta test data: |██████████|100%



... for final ensemble...


Generating meta training data: |██████████|100%
Training final base predictors: |██████████|100%




Training base predictors on MRI ROI: Volume (Cortical Parcellation)...
        
... for ensemble performance analysis...


Generating meta training data: |██████████|100%
Generating meta test data: |██████████|100%



... for final ensemble...


Generating meta training data: |██████████|100%
Training final base predictors: |██████████|100%




Training base predictors on MRI ROI: Surface Area...
        
... for ensemble performance analysis...


Generating meta training data: |██████████|100%
Generating meta test data: |██████████|100%



... for final ensemble...


Generating meta training data: |██████████|100%
Training final base predictors: |██████████|100%




Training base predictors on MRI ROI: Cortical Thickness Average...
        
... for ensemble performance analysis...


Generating meta training data: |██████████|100%
Generating meta test data: |██████████|100%



... for final ensemble...


Generating meta training data: |██████████|100%
Training final base predictors: |██████████|100%




Training base predictors on MRI ROI: Cortical Thickness Standard Deviation...
        
... for ensemble performance analysis...


Generating meta training data: |██████████|100%
Generating meta test data: |██████████|100%



... for final ensemble...


Generating meta training data: |██████████|100%
Training final base predictors: |██████████|100%


Analyzing ensembles: |██████████|100%
Training final meta models: |██████████|100%


In [18]:
performance_frame

undersampling  oversampling
0       0.823529      0.813725
1       0.361111      0.610561
2       0.278816      0.325397
3       0.370370      0.297101

In [23]:
performance_frame = performance_frame.set_index(pd.Index(metrics))
performance_frame

undersampling  oversampling
accuracy        0.823529      0.813725
precision       0.361111      0.610561
recall          0.278816      0.325397
f1              0.370370      0.297101